### ISSUE TITLE
False positive on importing typeguard.importhook

### ISSUE THREAD

**Issue by jolaf** [PyCQA/pylint#5265]
### Bug description

from typeguard import importhook
print(importhook)

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
test.py:1: [E0611(no-name-in-module), ] No name 'importhook' in module 'typeguard'
```

### Expected behavior

No error should be reported as the file is valid:

$ python3 test.py
<module 'typeguard.importhook' from '/home/user/.local/lib/python3.8/site-packages/typeguard/importhook.py'>

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.4
Python 3.8.10 (default, Sep 28 2021, 16:10:42) 
[GCC 9.3.0]
```

### OS / Environment

$ uname -a
Linux ubuntu 5.4.0-89-generic #100-Ubuntu SMP Fri Sep 24 14:50:10 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

$ cat /etc/issue
Ubuntu 20.04.3 LTS \n \l

### Additional dependencies

typeguard==2.13.0

**Comment by jacobtylerwalls** [comment]
I can reproduce on Python 3.8.10, but not on 3.10.1.

Reproduces
```
pylint 2.13.0-dev0
astroid 2.9.3
Python 3.8.10 (v3.8.10:3d8993a744, May  3 2021, 08:55:58) 
[Clang 6.0 (clang-600.0.57)]
```

Does not reproduce
```
pylint 2.13.0-dev0
astroid 2.9.3
Python 3.10.1 (v3.10.1:2cd268a3a9, Dec  6 2021, 14:28:59) [Clang 13.0.0 (clang-1300.0.29.3)]
```
